### PR TITLE
Document contributing guidelines 📝

### LINKED ISSUE #35: Document contribution guidelines 📝
- [x] Commit message conventions
- [x] Branch name conventations
- [x] Tag names (versioning scheme `YY.MINOR.MICRO`)
- [x] Milestones for issues and pull requests
- [x] Labels for issues and pull requests

### PR DESCRIPTION
**Pull request by hackebrot**
Resolve #35

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+```text
+Add parameterized unit test for sub2 validation
+
+The unit test uses a hyphenated parameter, a parameter that exceeds the maximum
+character count, and a parameter using emoji characters.
+```
+
+### Tag names
+
+Versions follow [Calendar Versioning][calver] using a `YY.MINOR.MICRO` scheme. 🗓
```
Comment (jrconlin) [inline .github/CONTRIBUTING.md:61]: Heh. I have... opinions... about this, but it's fine. They're just opinions.

**Review by jrconlin — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (hackebrot) [reply .github/CONTRIBUTING.md:61]: Ha! I was wondering if you would comment about this convention. 😅

Comment (jrconlin) [reply .github/CONTRIBUTING.md:61]: FWIW, I'm ok with this sort of thing for "stand alone" sort of packages. (e.g. a finalized application or OS) The reason is that you don't generally have subtle interactions that folks not intimately familiar with the packages have to be concerned about. Libraries or inter-operative applications do have to communicate breaking changes to users, and the best way to do that is via traditional semantic versioning. 

I might argue that the integration test version should tie to the associated version of contile (basically, "the last significant version of contile"). This would provide pressure that any notable change to the test operation should require a significant version change to contile as well. That way there's a cooperative check in place. 

The down side is that it would push localized versioning pretty far down (e.g. "0.4.1-210801" or whatever) and I'm not sure that's very useful. Granted, I *have* seen worse, more opaque, versions out there, so there's also that.

Comment (hackebrot) [reply .github/CONTRIBUTING.md:61]: Thank you for sharing your perspective on this, @jrconlin! I appreciate that.

I agree that it's critical packages and libraries communicate backward-incompatible changes to consumers. In my experience, projects that have claimed to strictly follow semantic versioning haven't always delivered on that promise, which is very unfortunate (API breaking changes in minor version bumps etc.). 😟 

Here's my motivation for using calendar versioning with a `YY.MINOR.MICRO` scheme for **contile-integration-tests**:

The most important factor is that we need to be able to easily tell if a version is newer than another one (this is straightforward with calver). The code in this repository is not released as a library and there are no packages that specify it as a dependency without pinning the version (we explicitly specify which Docker image tag to pull in the contile CI workflow). Totally agree that the integration test version correlates to the contile version, but not every contile release will require a new version of the integration tests. My concern is that forcing the integration tests to use the same version identifier may get confusing in the future and I don't think that's necessary.

In summary, my main goal is to make it straightforward for us to release new versions of the integration test Docker images without spending too much time on whether something constitutes a breaking change. Instead, every change to the CI workflow in contile needs to be carefully considered and discussed on the corresponding pull request. Also, I'm still open to moving this code into the contile repository which will dispense with the need for separate version identifiers for the integration tests (@mythmon and I will be discussing this for Merino this week).

Again, I appreciate your input! Thank you! 🙇🏻‍♂️

Comment (jrconlin) [reply .github/CONTRIBUTING.md:61]: Yep, that's why I don't consider my opinion on this one a blocking change. I know you've thought this out clearly and have really good reasons, and frankly, your opinion wins ;) . 

Thanks for hearing me out, presenting your case so well, and putting up with me. 

As for moving the code into the contile repo, aside from some careful work around building the docker images, there shouldn't be a huge issue.

**Review by ashrivastava-qa — APPROVED** [review]
This was very informative! Thanks @jrconlin and @hackebrot for this and I think we can go ahead with what Raphael suggested, in this scenario! :)

**Comment by hackebrot** [comment]
Thank you for your reviews @jrconlin and @ashrivastava-qa! Much appreciated. 📝
